ui/src/App.js

```diff
@@ -3,25 +3,30 @@ import { Button, Box, Typography, AppBar, Toolbar } from '@material-ui/core';
 import LandingPage from './LandingPage'
 import SurveyDetails from './SurveyDetails'
 import CollectionExerciseDetails from './CollectionExerciseDetails'
-import CaseSearch from './CaseSearch';
 import CaseDetails from './CaseDetails';
+import SurveyCaseSearch from './SurveyCaseSearch'
 
 class App extends Component {
   state = {
     selectedSurveyId: null,
     selectedSurveyName: '',
+    collectionExercises: [],
```

Comment:
I regret suggesting this now... I don't think it's great to be shoving stuff into the top level state, just to avoid a bit of chattiness with the backend. I preferred it the way you had it, having thought about it and seen that you've had to do this to make it work... but anyway, it's fine to leave it, but I regret this additional complexity

Reply:
I could put it back without too much difficulty. My issue with it then was that without refactoring in a different way, it needed the get collection exercises function copy/pasted which also felt ugly, I couldn't immediately see a good pattern for factoring out common functions?

Reply:
I'll just put it back for now then to avoid cluttering up the top level state with premature optimisation.